README.md

```diff
@@ -1 +1,81 @@
 # Immutable Web Apps
+
+## Introduction
+
+_Immutable Web Applications_ is a framework-agnostic methodology for building static [single-page application](https://en.wikipedia.org/wiki/Single-page_application) that:
+
+- Minimizes risk and complexity of live releases.
+- Simplifies and maximizes caching.
+- Minimizes the need for servers and systems administration.
+- Enables continuous delivery through simple, flexible, atomic deployments.
+
+## Principles
+
+The methodology is based on the principles of ___strictly separating___:
+
+- Configuration from code.
+- Release tasks from build tasks.
+- Dynamic content from static content.
+
+These principles are inspired by [The Twelve-Factor App](https://12factor.net/).
+
+## Concepts
+
+The following concepts define the core requirements for an _Immutable Web Application_. They are framework and infrastructure agnostic.
+
+### _Permabundles_
+
+Permabundles are the static assets that make up a single-page application.
+
+- They must be available at a [permalink](https://en.wikipedia.org/wiki/Permalink) that is uniquely versioned or fingerprinted and _independent of the web application environment(s)_.
+
+- They must be configured for long-term caching. (`cache-control: public, max-age=31536000, immutable`)
+
+- __They must not contain any environment-specific configuration.__
+
+### `index.html` <span style="font-size:larger;">___is___</span> configuration
+
+ An `index.html` exists for each deployment environment.
+
+- They must contain fully-qualified references to the _permabundle_ assets.
+
+- They must define all environment configuration on the global javascript scope.
```

Comment:
Given how Ember manages configuration via meta tag, I wonder if it makes sense to be less prescriptive about how the configuration is defined on the page?

Reply:
Agreed

Reply:
I also reordered the points to make them symmetrical between the two concepts, which I think demonstrates their relationship nicely.